### DISCUSSION TITLE
Any estimated release date or plans?

### DISCUSSION THREAD

**Discussion by borisdelev** [phalcon/phalcon#415] [General] [answered]
I know its hard to tell, but... is any possible release date of v6 or plan schedule?

PS: I was thinking to help in development, but I became a father with a boy :) anyway - i am still consider code supporting for mini stuff

**Comment by niden** [comment, marked as answer]
Congratulations!!! That is wonderful news.

Not sure yet is the answer. We are working to get the remaining namespaces converted and then we also have the 3 areas where C code has to be converted to PHP (annotations, phql and volt). @Jeckerson has been working on Volt so that should be ready pretty soon.

I have been battling with the `Mvc\Router` last weekend and once that is done I will move to other areas of it. Once `Mvc` is completed, the rest is not very difficult to address. 

The hope is to have everything ready before the end of the year but we will see how it goes.

**Reply by borisdelev** [reply]
Thanks- sounds awesome!
You done great work u know? Admire and keep going! I will support whatever i can.

May the force be with u!

**Reply by qmegas** [reply]
> The hope is to have everything ready before the end of the year but we will see how it goes.

Can you please clarify which year are you talking about? Since you wrote it in 2023 and now is 2025 and we still does not have v6 released.

**Reply by niden** [reply]
Year there was 2024. Clearly we did not reach that goal. We have only finished Volt and Phql is underway.

**Reply by borisdelev** [reply]
Well... Its understandable. Anyway- still v6 happening?
Ps: I wish I could help but no time with 2 kids 😔

**Reply by niden** [reply]
Yep absolutely. We are working on PHP 8.4 support, then PHQL parser and Annotations parser. After that we will release v6 alpha

**Reply by borisdelev** [reply]
Awesome.l!!! I just do donation but if there is still a way to help... I will.

**Reply by HalleyUK** [reply]
Hi Niden,
Any high-level idea on the release plan of v6, please? We are running on 5.8 currently, not sure if 5.8 will be maintained until a stable v6 is released.

**Reply by bekonRymkhanov** [reply]
hi @niden, how can you describe v6 progress?
really interesting to know about release date

**Reply by niden** [reply]
> Hi Niden, Any high-level idea on the release plan of v6, please? We are running on 5.8 currently, not sure if 5.8 will be maintained until a stable v6 is released.

5.10 is out with some bug fixes, so there is maintenance for the v5 series and will be ongoing until v6 is released (and more), so you are good there. The v6 package is ready-ish. The PHQL parser is missing from it, which of course for many of us is the core of the database operations, therefore v6 cannot be used as is. Once we have the PHQL parser done and tested we will release the first alpha and work from there.

I have created a series of videos on our youtube channel on how to use v6 with an API. That implementation uses the Data Mapper component of Phalcon instead of the Models. If your application can use the Data Mapper, then you can use v6 today and get it using composer.

**Reply by niden** [reply]
> hi @niden, how can you describe v6 progress? really interesting to know about release date

See above reply. 

Also forgot to mention that we have an implementation of PHQL done already but we need to thoroughly test it before it is released, since as I mentioned it is the core of most applications (the db operations)

**Reply by GrzegorzDrozd** [reply]
Question to understand better: do you plan to migrate Models also/eventually or will v6 be DataMapper only? If yes then this is very very bad news for me and my 400+ models :( It means that I will not be able to use v6 at all :/

**Reply by niden** [reply]
> Question to understand better: do you plan to migrate Models also/eventually or will v6 be DataMapper only? If yes then this is very very bad news for me and my 400+ models :( It means that I will not be able to use v6 at all :/

No actually quite the opposite. We NEED/WANT to migrate the models so that v6 can be a simple replacement for existing applications, and this is why the delay in releasing. 

If people do not use models or have a handful of them and want to migrate to Data Mapper, then they can use v6 right now.

**Comment by GrzegorzDrozd** [comment]
Hi @niden 

I cannot find an answer to this: will it require extension to run? Or will it work with regular php and detect extension loaded to speed things up? 

Thanks and keep up with the good work :)

**Reply by niden** [reply]
Phalcon v6 will be a pure PHP framework. All you will need will be a `composer require` command.

We will however offer an accompanying extension, which can also be loaded and will interact with the PHP installation to provide higher performance in areas that are inherently slow. Which areas? I have no idea at the moment but the plan is to run benchmarks on all of the code to find out which parts have the highest CPU cycles.

**Comment by ffaaddiill** [comment]
Hi, v6 is a great news !
Does phalcon v6 have a generator to generate the backend structure like Admin dashboard including its User Role or something like Access Control management ? So we can do something faster.

**Reply by niden** [reply]
Initially no, there will be no scaffolding. We can certainly extend devtools to cater for that once v6 is released.

**Comment by 123batman** [comment]
Hello @niden 
Do you have an update on how much longer until v6 is ready?

I'm waiting for v6 to use in a project. Should I expect end of the month or end of the year? Or even longer? I checked the repo and see steady progress but it is hard for me to tell how much longer it is going to take.

Thank you for your work :)

**Reply by extreme64** [reply]
..year

**Reply by niden** [reply]
I am hoping not a year. We are currently trying to get the PHQL parser done as I mentioned above. Once that is done the package is ready for testing and then release. Unfortunately I cannot give a more accurate estimate. Aiming to get this out this year, hopefully first or second quarter.
